driver-kotlin-sync/src/main/kotlin/com/mongodb/kotlin/client/AggregateIterable.kt

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kotlin.client
+
+import com.mongodb.ExplainVerbosity
+import com.mongodb.client.AggregateIterable as JAggregateIterable
+import com.mongodb.client.model.Collation
+import java.util.concurrent.TimeUnit
+import org.bson.BsonValue
+import org.bson.Document
+import org.bson.conversions.Bson
+
+/**
+ * Iterable like implementation for aggregate operations.
+ *
+ * @param T The type of the result.
+ * @see [Aggregation command](https://www.mongodb.com/docs/manual/reference/command/aggregate)
+ */
+public class AggregateIterable<T : Any>(private val wrapped: JAggregateIterable<T>) : MongoIterable<T>(wrapped) {
+    /**
+     * Sets the number of documents to return per batch.
+     *
+     * @param batchSize the batch size
+     * @return this
+     * @see [Batch Size](https://www.mongodb.com/docs/manual/reference/method/cursor.batchSize/#cursor.batchSize)
+     */
+    public override fun batchSize(batchSize: Int): AggregateIterable<T> = apply { wrapped.batchSize(batchSize) }
```

Comment:
You imply that wrapped.batchSize() returns wrapped. As JAggregateIterable is an interface, this is not guaranteed. (except if the javadoc mention is a contract)

Reply:
This is the contract - implementations have to return themselves (this) for these methods.

Reply:
ok